tests/scaletest/scale_test.go

```diff
@@ -1080,99 +1098,70 @@ func TestMain(t *testing.M) {
 }
 
 func TestSecureParallelCreationUpdationDeletionWithoutReboot(t *testing.T) {
-	SetupForTesting(t)
-	ingressType = SECURE
-	CreateIngressesParallel(t, numOfIng, initialNumOfPools)
-	UpdateIngressesParallel(t, numOfIng)
-	DeleteIngressesParallel(t, numOfIng, initialNumOfPools)
+	testIngressTypeWithReboot(t, REBOOTNONE, SECURE)
 }
 
 func TestSecureParallelCreationUpdationDeletionWithAkoReboot(t *testing.T) {
-	SetupForTesting(t)
-	ingressType = SECURE
-	CreateIngressParallelWithAkoReboot(t)
-	UpdateIngressParallelWithAkoReboot(t)
-	DeleteIngressParallelWithAkoReboot(t)
+	testIngressTypeWithReboot(t, REBOOTAKO, SECURE)
 }
 
 func TestSecureParallelCreationUpdationDeletionWithNodeReboot(t *testing.T) {
-	SetupForTesting(t)
-	ingressType = SECURE
-	CreateIngressParallelWithNodeReboot(t)
-	UpdateIngressParallelWithNodeReboot(t)
-	DeleteIngressParallelWithNodeReboot(t)
+	testIngressTypeWithReboot(t, REBOOTNODE, SECURE)
 }
 
 func TestSecureParallelCreationUpdationDeletionWithControllerReboot(t *testing.T) {
-	SetupForTesting(t)
-	ingressType = SECURE
-	CreateIngressParallelWithControllerReboot(t)
-	UpdateIngressParallelWithControllerReboot(t)
-	DeleteIngressParallelWithControllerReboot(t)
+	testIngressTypeWithReboot(t, REBOOTCONTROLLER, SECURE)
 }
 
 func TestMultiHostParallelCreationUpdationDeletionWithoutReboot(t *testing.T) {
-	SetupForTesting(t)
-	ingressType = MULTIHOST
-	CreateIngressesParallel(t, numOfIng, initialNumOfPools)
-	UpdateIngressesParallel(t, numOfIng)
-	DeleteIngressesParallel(t, numOfIng, initialNumOfPools)
+	testIngressTypeWithReboot(t, REBOOTNONE, MULTIHOST)
 }
 
 func TestMultiHostParallelCreationUpdationDeletionWithAkoReboot(t *testing.T) {
-	SetupForTesting(t)
-	ingressType = MULTIHOST
-	CreateIngressParallelWithAkoReboot(t)
-	UpdateIngressParallelWithAkoReboot(t)
-	DeleteIngressParallelWithAkoReboot(t)
+	testIngressTypeWithReboot(t, REBOOTAKO, MULTIHOST)
 }
 
 func TestMultiHostParallelCreationUpdationDeletionWithNodeReboot(t *testing.T) {
-	SetupForTesting(t)
-	ingressType = MULTIHOST
-	CreateIngressParallelWithNodeReboot(t)
-	UpdateIngressParallelWithNodeReboot(t)
-	DeleteIngressParallelWithNodeReboot(t)
+	testIngressTypeWithReboot(t, REBOOTNODE, MULTIHOST)
 }
 
 func TestMultiHostParallelCreationUpdationDeletionWithControllerReboot(t *testing.T) {
-	SetupForTesting(t)
-	ingressType = MULTIHOST
-	CreateIngressParallelWithControllerReboot(t)
-	UpdateIngressParallelWithControllerReboot(t)
-	DeleteIngressParallelWithControllerReboot(t)
+	testIngressTypeWithReboot(t, REBOOTCONTROLLER, MULTIHOST)
+}
+
+func TestAllIngressParallelCreationUpdationDeletionWithoutReboot(t *testing.T) {
+	ingTypes := []string{INSECURE, SECURE, MULTIHOST}
+	testMultipleIngressTypeWithReboot(t, REBOOTNONE, ingTypes)
+}
+func TestAllIngressParallelCreationUpdationDeletionWithAkoReboot(t *testing.T) {
+	ingTypes := []string{INSECURE, SECURE, MULTIHOST}
+	testMultipleIngressTypeWithReboot(t, REBOOTAKO, ingTypes)
```

Comment:
As discussed, the `REBOOTAKO` should be passed as a pointer. And so does `REBOOTNODE` and `REBOOTCONTROLLER` in the following testcases.